pkg/cloud/services/securitygroup/securitygroups.go

```diff
@@ -363,6 +368,53 @@ func (s *Service) describeSecurityGroupsByName() (map[string]infrav1.SecurityGro
 	return res, nil
 }
 
+// revokeIngressAndEgressRulesFromVPCDefaultSecurityGroup revokes ingress and egress rules from the VPC default security group.
+// The VPC default security group is created by AWS and cannot be deleted.
+// But we can revoke all ingress and egress rules from it to make it more secure. This security group is not used by CAPA.
+func (s *Service) revokeIngressAndEgressRulesFromVPCDefaultSecurityGroup() error {
+	if s.scope.VPC().SecureDefaultVPCSecurityGroup {
+		securityGroups, err := s.EC2Client.DescribeSecurityGroupsWithContext(context.TODO(), &ec2.DescribeSecurityGroupsInput{
+			Filters: []*ec2.Filter{
+				filter.EC2.VPC(s.scope.VPC().ID),
+				filter.EC2.SecurityGroupName("default"),
+			},
+		})
+		if err != nil {
+			return errors.Wrapf(err, "failed to find default security group in vpc %q", s.scope.VPC().ID)
+		}
+		defaultSecurityGroupID := *securityGroups.SecurityGroups[0].GroupId
+		s.scope.Debug("Removing ingress and egress rules from default security group in VPC", "defaultSecurityGroupID", defaultSecurityGroupID, "vpc-id", s.scope.VPC().ID)
+
+		ingressRules := infrav1.IngressRules{
+			{
+				Protocol:               infrav1.SecurityGroupProtocolAll,
+				FromPort:               -1,
+				ToPort:                 -1,
+				SourceSecurityGroupIDs: []string{defaultSecurityGroupID},
+			},
+		}
+		err = s.revokeSecurityGroupIngressRules(defaultSecurityGroupID, ingressRules)
+		if awserrors.IsIgnorableSecurityGroupError(errors.Cause(err)) != nil {
```

Comment:
The ingress and egress rules are removed successfully, but on the next reconciliations, I keep getting
```I1221 17:25:06.920228       1 securitygroups.go:386] "Removing ingress and egress rules from default security group in VPC" controller="awscluster" controllerGroup="infrastructure.cluster.x-k8s.io" controllerKind="AWSCluster" AWSCluster="org-giantswarm/jose4" namespace="org-giantswarm" name="jose4" reconcileID="7cee43d6-5cb8-4509-ad48-47e3c27b1dff" cluster="org-giantswarm/jose4" defaultSecurityGroupID="sg-09b4f9143ee0aeda4" vpc-id="vpc-0480fd3a1f02bc76b"```

What would be the best way to unwrap the error so that it's ignored? I figured that we don't want to do anything if the ingress or egress rules were already removed.

Reply:
We can use the same logic as `func (s *Service) ReconcileSecurityGroups`: first read existing rules with `s.getSecurityGroupIngressRules` and then reconcile the differences.

Reply:
I don't think it makes sense in this case, because all we want is to delete the existing rules. We don't care what's inside, or "the differences". We want the group to have no rules at all.